updates to support 1000-node SPS scaling test using the IDPS SRL pipeline

pymonger opened the pull request:
## Purpose
The purpose of this PR is to provide updates to SPS deployment, configuration and DAGs that will allow it to support a large scale test (1000 worker nodes).

## Proposed Changes
- [ADD] update tasks in EDRgen, RDRgen and vic2png DAGs with `weight_rule` and `priority_weight` so that large scale processing is not bottlenecked by the default priority scheme
- [CHANGE] published EDRgen, RDRgen and vic2png multi-arch docker images to ECR and updated KPOs to pull from it (during large scale testing, pulling images from dockerhub will quickly run into rate limiting error)
- [CHANGE] set `retries` to `3` for all KPO calls to improve dagRun resiliency
- [CHANGE] change KPO instance types to minimum instance type that can handle `c6i.large` PGE requirements
- [CHANGE] instantiate KPO instances as `spot` for cost savings
- [CHANGE] tuned various parameters in `airflow/helm/values.tmpl.yaml` that were required in order to perform the 1000-node SPS scaling test; the updated values were determined as a result of iterating over the large scale test many times and making these tweaks to resolve various bottlenecks
- [CHANGE] increased DB instance size to handle the traffic generated by the 1000-node scaling test
- [ADD] DAGs for evaluating readiness of the EDRgen, RDRgen, and vic2png jobs
- [ADD] added 2 new DAGs to aid in housecleaning of airflow dagRuns

## Issues
https://github.com/unity-sds/unity-sps/issues/285

## Testing
Video of the large scaling test can be viewed here: https://drive.google.com/file/d/1CZ1U0AOUy8bzm9HDWlq6YUVH75InzyEG/view?usp=drive_link.

pymonger committed:
add cleanup utility DAGs

pymonger committed:
minimal instance type for edrgen, rdrgen and vic2png

pymonger committed:
Merge branch 'develop' into srl-idps-scaling

pymonger committed:
Merge branch 'develop' into srl-idps-scaling

pymonger committed:
Merge branch 'develop' into srl-idps-scaling

pymonger committed:
updates for large scale testing

pymonger committed:
Merge branch 'develop' into srl-idps-scaling

pymonger committed:
add owner

pymonger committed:
Merge branch 'srl-idps-scaling' of github.com:pymonger/unity-sps into srl-idps-scaling

pymonger committed:
set owner correctly

LucaCinquini reviewed (changes requested):
Hi @pymonger : the changes look good, but I have the following comments:

o Do we still need the db_cleanup.py DAG since we now have a similar DAG: https://github.com/unity-sds/unity-sps/blob/develop/airflow/dags/db_cleanup_dag.py? I would remove this.

o Do we really need the delete_dag.py since it turns out Airflow allows you to delete a DAG from the UI? Upper right corner. I would remove this.

o I suggest changing 'values.tmpl.yaml' to another name like 'values_high_load.tmpl.yaml' so that it is not used by default. An administrator would need to explicitely override the default value to use the high load configuration.

o Also, is there some associated GitBook documentation?

Thanks very much!

pymonger committed:
Merge branch 'develop' into srl-idps-scaling

pymonger committed:
remove unneeded dags; create separate helm values template for high load

pymonger committed:
make which airflow helm values template a configurable terraform variable

pymonger commented:
@LucaCinquini: i've updated the PR to allow the deployment engineer to specify the helm values template via terraform variable. Also, i've updated the procedure to include this choice. You should've gotten a request for review here: https://app.gitbook.com/o/xZRqGQeQXJ0RP4VMj7Lq/s/UMIRhLdbRQTvMWop8Il9/~/changes/746/developer-docs/science-processing/docs/admin-guide/sps-deployment-with-terraform

Let me know if there's anything else.
